File: web/src/layout/SegmentedControlDemoXml.ts
//
// SegmentedControlDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId, startupAutoResize } from 'butterfly/dist/views/html'
import { customViewSetDelegate } from 'butterfly/dist/views/CustomView'
import htmlForDefault from './segmented_control_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.SegmentedControlDemoXml
export class SegmentedControlDemoXml {
    xmlRoot!: HTMLElement
    tabs!: HTMLDivElement
    tabs2!: HTMLDivElement
    loadHtmlString(): string {
        return htmlForDefault
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString())
        this.xmlRoot = view
        this.tabs = getViewById<HTMLDivElement>(view, "tabs")
        this.tabs2 = getViewById<HTMLDivElement>(view, "tabs2")
        return view
    }
}

File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/SegmentedControlDemoXml.kt
//
// SegmentedControlDemoXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class SegmentedControlDemoXml {

    lateinit var tabs: com.google.android.material.tabs.TabLayout
    lateinit var tabs2: com.google.android.material.tabs.TabLayout
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.segmented_control_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        tabs = view.findViewById<com.google.android.material.tabs.TabLayout>(R.id.tabs)
        tabs2 = view.findViewById<com.google.android.material.tabs.TabLayout>(R.id.tabs2)
        
        
        
        return view
    }
}